lib/IO/Socket/Async/SSL.pm6

```diff
@@ -95,6 +111,10 @@ class IO::Socket::Async::SSL {
     has $.enc;
     has $.insecure;
     has $!host;
+    has $!alpn;
+    # We need a non-attribute variable to use it during selection
+    my $server-protocols;
```

Comment:
If there are two concurrent connections ongoing, then we'll end up with a race between them, I think. Better is to nest the sub inside of the method, making it a closure, and then it will be possible to access the attributes.